src/modal/modal-manager.ts

```diff
@@ -68,19 +63,33 @@ export class ModalManager {
     }
   }
 
-  hideModal(): void {
-    if (this.modalElement) {
-      this.modalElement.modalHidden = true;
-      this.unregisterRootClickListener();
+  removeModal(): void {
+    const modal = this.getModal();
+    modal.modalHidden = true;
+    this.unregisterRootClickListener();
+    modal.remove();
+    this.modalElement = null;
+  }
+
+  private getModal(): CaptureEyeModal {
+    if (!this.modalElement) {
+      this.modalElement = document.createElement('capture-eye-modal');
+      this.modalElement.addEventListener('remove-capture-eye-modal', () => {
+        this.removeModal();
+      });
+      document.body.appendChild(this.modalElement);
+      return this.modalElement;
     }
+    return this.modalElement;
   }
 
   private updateModalAsset(
     assetModel: AssetModel | undefined,
     setAsLoaded: boolean
   ) {
-    if (!this.modalElement || !assetModel) return;
-    this.modalElement.updateAsset(assetModel, setAsLoaded);
+    const modal = this.getModal();
```

Comment:
how about keeping the original approach—checking if this.modalElement exists instead of recreating it when it doesn't? This way, we avoid unnecessary recreation, especially since this method might be triggered after a network query and the close button has already been clicked.